lntest/itest/lnd_channel-funding_fund_max_test.go

```diff
@@ -0,0 +1,277 @@
+	// subtracting the reserved amount to pay for a commitment transaction.
+	carolBalance btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+}
+
+func testChannelFundMax(net *lntest.NetworkHarness, ht *harnessTest) {
+	ctxb := context.Background()
+
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	carol := net.NewNode(
+		ht.t, "Carol", nil,
+	)
+	defer shutdownAndAssert(net, ht, carol)
+
+	dave := net.NewNode(
+		ht.t, "Dave", nil,
+	)
+	defer shutdownAndAssert(net, ht, dave)
+
+	// Ensure both sides are connected so the funding flow can be properly
+	// executed.
+	net.EnsureConnected(ht.t, carol, dave)
+
+	// Helper function that returns the fee estimate used for a tx
+	// with the given number of inputs and the optional change output.
+	// This matches the estimate done by the wallet.
+	fundingFee := func(numInput int, change bool) btcutil.Amount {
+		var weightEstimate input.TxWeightEstimator
+
+		// All inputs.
+		for i := 0; i < numInput; i++ {
+			weightEstimate.AddP2WKHInput()
+		}
+
+		// The multisig funding output.
+		weightEstimate.AddP2WSHOutput()
+
+		// Optionally count a change output.
+		if change {
+			weightEstimate.AddP2WKHOutput()
+		}
+
+		totalWeight := int64(weightEstimate.Weight())
+		return feeRate.FeeForWeight(totalWeight)
+	}
+
+	// Helper closure to assert the proper response to a channel balance RPC.
+	checkChannelBalance := func(node lnrpc.LightningClient,
+		amount btcutil.Amount) error {
+
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		response, err := node.ChannelBalance(
+			ctxt, &lnrpc.ChannelBalanceRequest{},
+		)
+		if err != nil {
+			ht.Fatalf("unable to get channel balance: %v", err)
+		}
+
+		balance := btcutil.Amount(response.Balance)
+		if balance != amount {
+			return fmt.Errorf("channel balance wrong: got (%v), "+
+				"want (%v)", balance, amount)
+		}
+
+		return nil
+	}
+
+	// Helper function to sweep funds from a node wallet.
+	var minerAddr btcutil.Address
+	sweepNodeWalletAndAssert := func(node *lntest.HarnessNode) {
+		// Send funds back to the miner node.
+		if minerAddr == nil {
+			var err error
+			minerAddr, err = net.Miner.NewAddress()
+			if err != nil {
+				ht.Fatalf(
+					"unable to create new miner address: %v",
+					err,
+				)
+			}
+		}
+
+		ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+		if _, err := carol.SendCoins(ctxt, &lnrpc.SendCoinsRequest{
+			Addr:    minerAddr.String(),
+			SendAll: true,
+		}); err != nil {
+			ht.Fatalf("unable to sweep coins from node %v: %v",
+				node.Name(), err)
+		}
+
+		// Ensures we don't leave any transaction in the mempool after sweeping.
+		mineBlocks(ht, net, 1, 1)
+
+		// Make sure all funds were swept.
+		checkChannelBalance(node, 0)
+	}
+
+	runTestCase := func(carol, dave *lntest.HarnessNode,
+		testCase *chanFundMaxTestCase) func(t *testing.T) {
+
+		return func(t *testing.T) {
```

Comment:
To avoid this nesting and the code within reaching over the 80 character limit, we shouldn't return a function here, but instead just define a function that takes `t, carol, dave, testCase` and then _call_ it in the test case loop below.

Reply:
Has been covered here: https://github.com/lightningnetwork/lnd/pull/6903